#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.35044153985650767 -0.9333196435446259 -0.0781355880296047 5.814189837386369
  position 0.9415339743573068 0.5246920334236671 1.7030044582347852
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0 0.075 0
  children [
    DEF SH_BODY Group {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.8 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.25 0.08 0.1
          }
        }
        Transform {
          translation 0.1075 -0.0575 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.9294117647058824 0.8313725490196079 0
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.0175
              }
            }
          ]
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.05 0 0.065
      }
      device [
        RotationalMotor {
          name "motor_right"
          maxVelocity 52.359877499999996
          maxTorque 0.59317
        }
        PositionSensor {
          name "sensor_right"
          resolution 0.005235983
        }
      ]
      endPoint Solid {
        translation -0.05 0 0.065
        rotation 1 0 0 1.5708
        children [
          DEF SH_WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.03
              radius 0.075
            }
          }
        ]
        boundingObject USE SH_WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.05 0 -0.065
      }
      device [
        RotationalMotor {
          name "motor_left"
          maxVelocity 52.3599
          maxTorque 0.59317
        }
        PositionSensor {
          name "sensor_left"
          resolution 0.00523598
        }
      ]
      endPoint Solid {
        translation -0.05 0 -0.065
        rotation 1 0 0 1.5707996938995745
        children [
          USE SH_WHEEL
        ]
        name "solid(1)"
        boundingObject USE SH_WHEEL
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.125 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.3333333333333333 0.3411764705882353 0.3254901960784314
            roughness 1
            metalness 0
          }
          geometry DEF Sen Box {
            size 0.01 0.01 0.02
          }
        }
      ]
      name "distance_sensor"
      boundingObject USE Sen
      lookupTable [
        0 0 0
        0.02 0 0
        5 1660 0
      ]
      resolution 1
    }
  ]
  boundingObject USE SH_BODY
  physics Physics {
  }
  controller "crazybot"
}
